### DISCUSSION TITLE
Clearer Less confusing History Section in Room settings

### DISCUSSION THREAD

**Discussion by jph168** [element-hq/element-meta#470] [Ideas]
#### What would you like to do?

instead of: 

Who Can Read History?
Changes to who can read history will only apply to future messages in this room. The visibility of existing history will be unchanged.

Members only (since the point in time of selecting this option)
Members only (since they were invited)
Members only (since they joined)

something more along the lines of:

Who can Read Room History?
This will only apply to future messages in this room for anyone joining. Room History will be unchanged for anyone in this room.

Members only (All previous Messages)
Members only (Messages since Invited)
Members only (Messages since Joined)

#### Why would you like to do it?

i believe it would make this a lot less confusing.

#### How would you like to achieve it?

Upstream with your help :)

**Comment by t3chguy** [comment]
Moving to meta as mobile has the same wording

**Comment by jwmh** [comment]
I personally favor the current wording as being less-confusing, vs the OP’s proposed wording. ::shrug:: fwiw

But I think the situation in general is a bit complicated.

Is there an easier way to make it less muddy in understanding?
